swagger/control/swagger.yaml

```diff
@@ -0,0 +1,152 @@
+      parameters:
+        - name: descriptor
+          in: body
+          schema:
+            $ref: "#/definitions/Descriptor"
+          required: true
+      consumes:
+        - application/json
+      responses:
+        200:
+          description: signals that the token was correctly erased.
+        404:
+          description: signals that no token associated to the given descriptor exists.
+        default:
+          description: contains an unexpected error.
+
+  /api/channel:
+    put:
+      operationId: put_channel
+      tags:
+        - control
+      description: |
+        updates the channel data associated with the descriptor.
+
+        If there is already a channel associated with the descriptor, the old channel is overwritten with the new one.
+      parameters:
+        - name: descriptor
+          in: body
+          schema:
+            $ref: "#/definitions/Descriptor"
+          required: true
+        - name: channel
+          in: body
+          schema:
+            $ref: "#/definitions/Channel"
+          required: true
+      consumes:
+        - application/json
+      responses:
+        200:
+          description: signals that the channel was accepted.
+        default:
+          description: contains an unexpected error.
+    delete:
+      operationId: delete_channel
+      tags:
+        - control
+      description: |
+        removes the channel associated with the descriptor.
+      parameters:
+        - name: descriptor
+          in: body
+          schema:
+            $ref: "#/definitions/Descriptor"
+          required: true
+      consumes:
+        - application/json
+      responses:
+        200:
+          description: signals that the channel was correctly erased.
+        404:
+          description: signals that no channel associated to the given descriptor was found.
+        default:
+          description: contains an unexpected error.
+
+definitions:
+  Token:
+    description: is a string authenticating the sender of an HTTP request.
+    type: string
+    example: RBbPYhmPXurT8nM5TAJpPOcHMaFkJblA62mr6MCvpF4oVa6cy
+
+  Descriptor:
+    description: uniquely identifies an instance.
+    type: string
+    example: client-1/pipeline-3
+
+  Entity:
+    description: contains the email address and optionally the name of an entity.
+    type: object
+    properties:
+      email:
+        type: string
+        example: name@domain.com
+      name:
+        type: string
+        example: John Doe
+    required:
+      - email
+
+
+  Channel:
+    description: contains the channel settings for messages received by an instance.
+    type: object
+    properties:
+      sender:
+        $ref: "#/definitions/Entity"
+      recipients:
+        type: array
+        items:
+          $ref: "#/definitions/Entity"
+      cc:
+        type: array
+        items:
+          $ref: "#/definitions/Entity"
+      bcc:
+        type: array
+        items:
+          $ref: "#/definitions/Entity"
```

Comment:
(Following the comment above) I'd include the token in the Channel definition here.